astrobee/survey_manager/survey_planner/tools/survey_planner/command_astrobee

```diff
@@ -0,0 +1,512 @@
+        self.pub_command = rospy.Publisher(self.ns + "/command", CommandStamped, queue_size=5)
+        while self.pub_command.get_num_connections() == 0:
+            rospy.loginfo("Waiting for subscriber to connect")
+            rospy.sleep(1)
+        self.unique_cmd_id = ""
+
+    def start_recording(self, bag_description):
+        # Arg is bagfile name description
+        arg1 = CommandArg()
+        arg1.data_type = CommandArg.DATA_TYPE_STRING
+        arg1.s = bag_description
+
+        cmd = CommandStamped()
+        cmd.header = Header(stamp=rospy.Time.now())
+        cmd.cmd_name = CommandConstants.CMD_NAME_START_RECORDING
+        cmd.cmd_id = "survey_manager" + str(rospy.Time.now().to_sec())
+        self.unique_cmd_id = cmd.cmd_id
+        cmd.cmd_src = "isaac fsw"
+        cmd.cmd_origin = "isaac fsw"
+        cmd.args = [arg1]
+
+        # Publish the CommandStamped message
+        result = self.publish_and_wait_response(cmd)
+        return result
+
+    def stop_recording(self):
+        cmd = CommandStamped()
+        cmd.header = Header(stamp=rospy.Time.now())
+        cmd.cmd_name = CommandConstants.CMD_NAME_STOP_RECORDING
+        cmd.cmd_id = "survey_manager" + str(rospy.Time.now().to_sec())
+        self.unique_cmd_id = cmd.cmd_id
+        cmd.cmd_src = "isaac fsw"
+        cmd.cmd_origin = "isaac fsw"
+
+        # Publish the CommandStamped message
+        result = self.publish_and_wait_response(cmd)
+        return result
+
+    def change_exposure(self, val):
+        #TBD
+        rospy.loginfo("Change exposure to " + str(val))
+        return 0
+
+    def change_map(self, map_name):
+        #TBD
+        rospy.loginfo("Change map to " + map_name)
+        return 0
+
+    def set_plan(self):
+        # TODO Add call to plan_pub.cc
+
+        cmd = CommandStamped()
+        cmd.header = Header(stamp=rospy.Time.now())
+        cmd.cmd_name = CommandConstants.CMD_NAME_SET_PLAN
+        cmd.cmd_id = "survey_manager" + str(rospy.Time.now().to_sec())
+        self.unique_cmd_id = cmd.cmd_id
+        cmd.cmd_src = "isaac fsw"
+        cmd.cmd_origin = "isaac fsw"
+        cmd.args = cmd_args
+
+        # Publish the CommandStamped message
+        result = self.publish_and_wait_response(cmd)
+        return result
+
+    def run_plan(self):
+        cmd = CommandStamped()
+        cmd.header = Header(stamp=rospy.Time.now())
+        cmd.cmd_name = CommandConstants.CMD_NAME_RUN_PLAN
+        cmd.cmd_id = "survey_manager" + str(rospy.Time.now().to_sec())
+        self.unique_cmd_id = cmd.cmd_id
+        cmd.cmd_src = "isaac fsw"
+        cmd.cmd_origin = "isaac fsw"
+
+        # Publish the CommandStamped message
+        result = self.publish_and_wait_response(cmd)
+        return result
+
+    def ack_callback(self, msg):
+        if self.ack_needed == True and msg.cmd_id == self.unique_cmd_id:
+            self.ack_msg = msg
+            self.ack_needed = False
+
+    def publish_and_wait_response(self, cmd):
```

Comment:
per exit codes standards, anything other than zero is failure, so I'm passing a non-zero code back if it fails. I want it to be a number instead of a bool because we're adding the exit_codes to evaluate overall success